[esx]/es_extended/client/main.lua

```diff
@@ -64,53 +64,58 @@ AddEventHandler('esx:playerLoaded', function(xPlayer, isNew, skin)
 			local RemoveAllPickupsOfType = RemoveAllPickupsOfType
 			local HideHudComponentThisFrame = HideHudComponentThisFrame
 			local PlayerId = PlayerId()
+			local ESXPlayerId = ESX.PlayerData.ped
 			local DisabledComps = {}
 			for i=1, #(Config.RemoveHudCommonents) do
 				if Config.RemoveHudCommonents[i] then
 					DisabledComps[#DisabledComps + 1] = i
 				end
-		 end
+		 	end
+			if Config.DisableHealthRegeneration then
+				SetPlayerHealthRechargeMultiplier(PlayerId, 0.0)
+			end
+
+			if Config.DisableNPCDrops then
+				AddEventHandler('gameEventTriggered', function(name, args)
+    				if name == 'CEventNetworkEntityDamage' then
+        				if args[6] == 1 then
+            				SetPedDropsWeaponsWhenDead(args[1], false)
+        				end
+    				end
+				end)
+			end
 			while true do 
 				local Sleep = true
 
-				if Config.DisableHealthRegeneration then
-					Sleep = false
-					SetPlayerHealthRechargeMultiplier(PlayerId, 0.0)
-				end
-
 				if Config.DisableWeaponWheel then
 					Sleep = false
 					BlockWeaponWheelThisFrame()
 					DisableControlAction(0, 37,true)
 				end
 
 				if Config.DisableAimAssist then
-					Sleep = false
-					if IsPedArmed(ESX.PlayerData.ped, 4) then
+					if IsPedArmed(ESXPlayerId, 4) then
+						Sleep = false
 						SetPlayerLockonRangeOverride(PlayerId, 2.0)
 					end
 				end
 
 				if Config.DisableVehicleRewards then
-					Sleep = false
-					DisablePlayerVehicleRewards(PlayerId)
+					local veh = GetVehiclePedIsEntering(ESXPlayerId)
+        			if veh ~= 0 then
+						Sleep = false
```

Comment:
This does not do anything except for delay the next call

Reply:
Removed delay @Gittified